src/service/eol/eol.svc.ts

```diff
@@ -62,41 +59,45 @@ export async function submitScan(model: SbomModel): Promise<ScanResult> {
  * processing and/or rendering.
  */
 export async function prepareRows({ components, purls }: SbomModel, scan: ScanResult): Promise<Line[]> {
-  const lines = purls
-    .map((purl): Line | null => {
-      const details = scan.components.get(purl);
-
-      if (!details) {
-        // In this case, the purl string is in the generated sbom, but the NES/XEOL api has no data
-        log.debug(`Unknown status: ${purl}.`);
-        return null;
-      }
+  const lines: Line[] = [];
+
+  for (const purl of purls) {
+    const details = scan.components.get(purl);
+
+    if (!details) {
+      // In this case, the purl string is in the generated sbom, but the NES/XEOL api has no data
+      log.debug(`Unknown status: ${purl}.`);
+      continue;
+    }
+
+    const { evidence } = components[purl];
+    const occ = evidence?.occurrences?.map((o) => o.location).join('\n\t - ');
+
+    const SHOW_OCCURRENCES = (process.env.SHOW_OCCURRENCES || 'false') === 'true';
+    const occurrences = SHOW_OCCURRENCES && Boolean(occ) ? `\t - ${occ}\n` : '';
 
-      const { evidence } = components[purl];
-      const occ = evidence?.occurrences?.map((o) => o.location).join('\n\t - ');
-      const occurrences = SHOW_OCCURRENCES && Boolean(occ) ? `\t - ${occ}\n` : '';
+    const { info } = details;
 
-      const { info } = details;
+    // Handle date deserialization from GraphQL
+    info.eolAt = typeof info.eolAt === 'string' && info.eolAt ? new Date(info.eolAt) : info.eolAt;
```

Comment:
```suggestion
    if (typeof info.eolAt === 'string' && info.eolAt) {
      info.eolAt = new Date(info.eolAt);
    }
```
Avoid unnecessary reassignment